src/client/opamCommands.ml

```diff
@@ -2199,9 +2199,24 @@ let repository cli =
           (OpamSwitch.to_string (OpamStateConfig.get_switch ()))
           (OpamRepositoryName.to_string name);
       `Ok ()
+    | Some `remove, [] ->
+      OpamConsole.error_and_exit `Not_found
+        "No repository name was given, \
+         please give at least one name in parameter"
     | Some `remove, names ->
       let names = List.map OpamRepositoryName.of_string names in
-      let rm = List.filter (fun n -> not (List.mem n names)) in
+      let seen_repos = ref [] in
+      let rm =
```

Comment:
Actually this is not even called when the repository is not attached to a switch but exists in the global state so it returns an error even though it did something.
```
$ ./opam repo -a
[...]
tmp          [...]
$ ./opam repo remove tmp -a
[ERROR] No repository named tmp
$ ./opam repo -a
[...]
[no tmp]
```

Reply:
now fixed